examples/tests/valid/struct_json_conversion.w

```diff
@@ -0,0 +1,217 @@
+  firstName: str;
+  lastName: str;
+  dob: Date;
+}
+
+struct Advisor extends Person {
+  employeeID: str;
+}
+
+struct Course {
+  name: str;
+  credits: num;
+}
+
+struct CourseResults {
+  course: Course;
+  grade: str;
+  dateTaken: Date;
+}
+
+struct Student extends Person {
+  enrolled: bool;
+  schoolId: str;
+  advisor: Advisor?;
+  enrolledCourses: Set<Course>?;
+  coursesTaken: Array<CourseResults>?;
+  additionalData: Json?;
+}
+
+// Student with no advisor and no empty course list
+let jStudent1 = {
+  firstName: "John",
+  lastName: "Smith",
+  enrolled: true,
+  schoolId: "s1-xyz",
+  dob: { month: 10, day: 10, year: 2005 },
+  enrolledCourses: []
+};
+
+let student1 = Student.fromJson(jStudent1);
+assert(student1.firstName == "John");
+assert(student1.lastName == "Smith");
+assert(student1.enrolled);
+assert(student1.schoolId == "s1-xyz");
+assert(student1.dob.month == 10);
+assert(student1.dob.day == 10);
+assert(student1.dob.year == 2005);
+
+// Student with an advisor and several courses
+let jStudent2 = {
+  advisor: {
+    firstName: "Tom",
+    lastName: "Baker",
+    dob: { month: 1, day: 1, year: 1983 },
+    employeeID: "emp123"
+  },
+  firstName: "Sally",
+  lastName: "Reynolds",
+  enrolled: false,
+  schoolId: "s2-xyz",
+  dob: { month: 5, day: 31, year: 1987},
+  enrolledCourses: [
+    { name: "COMP 101", credits: 2 },
+    { name: "COMP 121", credits: 4 },
+  ],
+  coursesTaken: [
+    {grade: "F", dateTaken: { month: 5, day: 10, year: 2021 }, course: { name: "COMP 101", credits: 2 }},
+    {grade: "D", dateTaken: { month: 5, day: 10, year: 2021 }, course: { name: "COMP 121", credits: 4 }},
+  ]
+};
+
+let student2 = Student.fromJson(jStudent2);
+assert(student2.firstName == "Sally");
+assert(student2.lastName == "Reynolds");
+assert(!student2.enrolled);
+assert(student2.schoolId == "s2-xyz");
+assert(student2.dob.month == 5);
+assert(student2.dob.day == 31);
+assert(student2.dob.year == 1987);
+
+if let enrolledCourses = student2.enrolledCourses {
+  let courses = enrolledCourses.toArray();
+  let s2Course1 = courses.at(0);
+  let s2Course2 = courses.at(1);
+  
+  assert(s2Course1.name == "COMP 101");
+  assert(s2Course1.credits == 2);
+  assert(s2Course2.name == "COMP 121");
+  assert(s2Course2.credits == 4);
+} else {
+  assert(false); // It should never hit this
+}
+
+// Create a student that has additional data (Json)
+let jStudent3 = {
+  enrolled: false,
+  schoolId: "w/e",
+  firstName: student2.firstName,
+  lastName: student2.lastName,
+  dob: {
+    month: 1, day: 1, year: 1959
+  },
+  additionalData: {
+    notes: "wow such notes",
+    legacy: false,
+    emergencyContactsNumbers: [
+      "123-345-9928",
+    ]
+  }
+};
+
+let student3 = Student.fromJson(jStudent3);
+
+if let additionalData = student3.additionalData {
+  let notes = additionalData.get("notes");
+  assert(notes == "wow such notes");
+} else {
+  assert(false); // shouldnt happen
+}
+
+// struct Student extends Person {
```

Comment:
Why is this commented out?

Reply:
was leftover debug, the struct definition is uncommented earlier in the file I removed this.